src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -209,10 +230,44 @@ class SidebarPanels extends React.Component<Props, State> {
         }
     }
 
+    getPanelOrder = (customPanel?: CustomSidebarPanel, shouldBoxAIBeDefaultPanel: boolean): string[] => {
+        const { index: insertIndex = 0, shouldBeDefaultPanel } = customPanel || {};
+        const customPanelId = customPanel && typeof customPanel.id === 'string' ? customPanel.id.trim() : '';
+        const hasBoxAICustomPanel = customPanelId === SIDEBAR_VIEW_BOXAI;
+
+        // Build base panel list without custom panel
+        const getBasePanels = () => {
+            if (!hasBoxAICustomPanel && shouldBoxAIBeDefaultPanel) {
+                return [SIDEBAR_VIEW_BOXAI, ...DEFAULT_SIDEBAR_VIEWS];
+            }
+            return hasBoxAICustomPanel ? DEFAULT_SIDEBAR_VIEWS : [...DEFAULT_SIDEBAR_VIEWS, SIDEBAR_VIEW_BOXAI];
+        };
```

Comment:
I don't think you should try to account for the boxAI logic directly in this component and instead handle it in the consuming application by either omitting or providing a customPanel and enabling or disabling using boxAI feature in the sidebar props

it'll simplify the overall logic and there will be less clean up work later when you remove boxAiSidebar